client/src/presenters/CoursePresenter.tsx

```diff
@@ -3,138 +3,156 @@ import { Course } from "../models/common";
 import { useEffect, useState } from "react";
 import SectionPresenter from "./SectionPresenter";
 import {
-  ContentContainer,
-  YellowPurpleGradiant,
-  sectionHeaderStyle,
+	ContentContainer,
+	YellowPurpleGradiant,
+	sectionHeaderStyle,
 } from "../constants/styleNames";
 import { Navbar } from "../shared/Navbar";
 
 export const CoursePresenter = () => {
-  const location = useLocation();
-  const { courseCode } = useParams();
-  const [course, setCourse] = useState<Course>();
-  const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+	const location = useLocation();
+	const { courseCode } = useParams();
+	const [course, setCourse] = useState<Course>();
+	const [isCollapsed, setIsCollapsed] = useState<boolean>(false);
+	const [loading, setLoading] = useState(true);
 
-  useEffect(() => {
-    if (courseCode != null) {
-      // we should expect the url to be in the format /[CourseName][CourseNumber], where the two are split via the first numeric character's appearance
-      var courseFaculty = "";
-      var courseNumber = "";
+	useEffect(() => {
+		if (courseCode != null) {
+			// we should expect the url to be in the format /[CourseName][CourseNumber], where the two are split via the first numeric character's appearance
+			var courseFaculty = "";
+			var courseNumber = "";
 
-      for (var i = 0; i < courseCode.length; i++) {
-        if (!isNaN(Number(courseCode[i]))) {
-          courseNumber = courseCode.slice(i);
-          break;
-        }
-        courseFaculty += courseCode[i];
-      }
-      const queryString = `http://localhost:8080/api/v1/courses?course-faculty=exact:${courseFaculty.toUpperCase()}&course-number=gte:${courseNumber}&course-number=lte:${courseNumber}`;
-      fetch(queryString)
-        .then((res) => res.json())
-        .then((res) => {
-          if (res && res.length > 0) setCourse(res[0] as Course);
-        });
-    }
-  }, [location.search, courseCode]);
+			for (var i = 0; i < courseCode.length; i++) {
```

Comment:
i think ur formatter might be set to 4 spaces while im using 2. can u change the spacing to 2 so we are consistent?